hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileReader.java

```diff
@@ -62,6 +64,7 @@
   // Scanner used to read individual keys. This is cached to prevent the overhead of opening the scanner for each
   // key retrieval.
   private HFileScanner keyScanner;
+  private final String keyField = HoodieMetadataPayload.SCHEMA_FIELD_ID_KEY;
```

Comment:
We could even stick this into the config object. But cannot do this hardcoding or referencing of metadata payload here.

Reply:
fixed.